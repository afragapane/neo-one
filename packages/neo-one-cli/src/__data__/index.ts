export * from './getClients';
export * from './getContracts';
