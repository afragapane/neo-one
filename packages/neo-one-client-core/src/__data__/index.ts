export * from './keys';
export * from './data';
