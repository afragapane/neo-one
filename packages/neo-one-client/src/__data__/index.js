/* @flow */
export { default as keys } from './keys';
export { default as transactions } from './transactions';
