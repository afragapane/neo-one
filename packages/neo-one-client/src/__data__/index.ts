export * from './keys';
export * from './data';
export * from './transactions';
export * from './factory';
