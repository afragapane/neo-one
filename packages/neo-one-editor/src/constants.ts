// tslint:disable-next-line export-name
export const METADATA_FILE = '/.$metadata';
