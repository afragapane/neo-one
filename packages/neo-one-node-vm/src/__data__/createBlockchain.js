/* @flow */
export default ({ contract }: {| contract?: Buffer |}): $FlowFixMe => {
  const blockchain = ({
    contract: {},
    storageItem: {},
    settings: {
      vm: {},
    },
    action: {},
    output: {},
  }: $FlowFixMe);
  if (contract != null) {
    blockchain.contract.get = jest.fn(() => Promise.resolve(contract));
  }

  const storage = {};
  blockchain.storageItem.add = jest.fn(async item => {
    storage[item.key.toString('hex')] = item;
  });
  blockchain.storageItem.update = jest.fn(async (item, { value }) => {
    storage[item.key.toString('hex')] = {
      ...item,
      value,
    };
  });
  blockchain.storageItem.get = jest.fn(async ({ key }) => {
    const item = storage[key.toString('hex')];
    if (item == null) {
      throw new Error('Item not found');
    }
    return item;
  });
  blockchain.storageItem.tryGet = jest.fn(
    async ({ key }) => storage[key.toString('hex')],
  );

  blockchain.action.add = jest.fn(() => Promise.resolve());

  return blockchain;
};
