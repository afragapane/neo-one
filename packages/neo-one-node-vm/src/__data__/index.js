/* @flow */
export { default as createBlockchain } from './createBlockchain';
export { default as keys } from './keys';
export * as transactions from './transactions';
export * as testUtils from './utils';
