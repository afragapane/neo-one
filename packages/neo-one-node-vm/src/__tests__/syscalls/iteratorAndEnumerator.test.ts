// tslint:disable no-object-mutation
import { StorageFlags, StorageItem } from '@neo-one/node-core';
import { AsyncIterableX } from '@reactivex/ix-es2015-cjs/asynciterable/asynciterablex';
import { BN } from 'bn.js';
import { runSysCalls, scriptAttributeHash, TestCase } from '../../__data__';
import { FEES } from '../../constants';
import {
  BooleanStackItem,
  BufferStackItem,
  IntegerStackItem,
  IteratorStackItem,
  StackItemIterator,
} from '../../stackItem';

// tslint:disable-next-line readonly-array
const testArray = [1, 2, 3];
const testIterator: ReadonlyArray<{ readonly key: IntegerStackItem; readonly value: IntegerStackItem }> = [
  { key: new IntegerStackItem(new BN(0)), value: new IntegerStackItem(new BN(1)) },
  { key: new IntegerStackItem(new BN(1)), value: new IntegerStackItem(new BN(2)) },
  { key: new IntegerStackItem(new BN(2)), value: new IntegerStackItem(new BN(3)) },
];
const testAsyncIterable = AsyncIterableX.from(testIterator);
const nextItem = new StorageItem({
  hash: scriptAttributeHash,
  key: Buffer.from('key', 'utf-8'),
  value: Buffer.from('val', 'utf-8'),
  flags: StorageFlags.None,
});

const SYSCALLS: readonly TestCase[] = [
  {
    name: 'System.Iterator.Create',
    result: [new IteratorStackItem(new StackItemIterator(testAsyncIterable[Symbol.asyncIterator]()))],
    args: [testArray],
    gas: FEES[400],
  },

  {
    name: 'System.Iterator.Key',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Iterator.Create',
            type: 'sys',
            args: [
              {
                type: 'calls',
                calls: [
                  {
                    name: 'NEWMAP',
                    type: 'op',
                  },

                  {
                    name: 'DUP',
                    type: 'op',
                  },

                  {
                    name: 'SETITEM',
                    type: 'op',
                    args: [Buffer.from('value2', 'utf8'), Buffer.from('key2', 'utf8')],
                  },
                ],
              },
            ],
          },

          {
            name: 'System.Iterator.Create',
            type: 'sys',
            args: [
              {
                type: 'calls',
                calls: [
                  {
                    name: 'NEWMAP',
                    type: 'op',
                  },

                  {
                    name: 'DUP',
                    type: 'op',
                  },

                  {
                    name: 'SETITEM',
                    type: 'op',
                    args: [Buffer.from('value1', 'utf8'), Buffer.from('key1', 'utf8')],
                  },
                ],
              },
            ],
          },

          {
            name: 'System.Iterator.Concat',
            type: 'sys',
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },
        ],
      },
    ],

    result: [new BufferStackItem(Buffer.from('key2', 'utf8'))],
    gas: FEES[400],
  },

  {
    name: 'System.Iterator.Key',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'SWAP',
            type: 'op',
            args: [
              {
                type: 'calls',
                calls: [
                  {
                    name: 'System.Enumerator.Next',
                    type: 'sys',
                    args: [
                      {
                        type: 'calls',
                        calls: [
                          {
                            name: 'DUP',
                            type: 'op',
                            args: [
                              {
                                type: 'calls',
                                calls: [
                                  {
                                    name: 'System.Storage.Find',
                                    type: 'sys',
                                    args: [
                                      {
                                        type: 'calls',
                                        calls: [
                                          {
                                            name: 'System.Storage.GetContext',
                                            type: 'sys',
                                          },
                                        ],
                                      },

                                      Buffer.alloc(1, 1),
                                    ],
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      },
    ],

    result: [new BufferStackItem(nextItem.key), new BooleanStackItem(true)],
    mockBlockchain: ({ blockchain }) => {
      blockchain.contract.get = jest.fn(async () => Promise.resolve({ manifest: { hasStorage: true } }));

      blockchain.storageItem.getAll$ = jest.fn(() => AsyncIterableX.of(nextItem));
    },
    gas: FEES[400],
  },

  {
    name: 'System.Enumerator.Next',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Enumerator.Create',
            type: 'sys',
            args: [[new BN(0)]],
          },
        ],
      },
    ],

    result: [new BooleanStackItem(true)],
    gas: FEES[1_000_000],
  },

  {
    name: 'System.Enumerator.Next',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Iterator.Create',
            type: 'sys',
            args: [
              {
                type: 'calls',
                calls: [
                  {
                    name: 'NEWMAP',
                    type: 'op',
                  },

                  {
                    name: 'DUP',
                    type: 'op',
                  },

                  {
                    name: 'SETITEM',
                    type: 'op',
                    args: [Buffer.from('value', 'utf8'), Buffer.from('key', 'utf8')],
                  },
                ],
              },
            ],
          },
        ],
      },
    ],

    result: [new BooleanStackItem(true)],
    gas: FEES[1_000_000],
  },

  {
    name: 'System.Enumerator.Next',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Enumerator.Create',
            type: 'sys',
            args: [[new BN(0)]],
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },
        ],
      },
    ],

    result: [new BooleanStackItem(false)],
    gas: FEES[1_000_000],
  },

  {
    name: 'System.Enumerator.Value',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Enumerator.Create',
            type: 'sys',
            args: [[new BN(1), new BN(2)]],
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },
        ],
      },
    ],

    result: [new IntegerStackItem(new BN(1))],
    gas: FEES[400],
  },

  {
    name: 'System.Enumerator.Next',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Enumerator.Create',
            type: 'sys',
            args: [[new BN(1)]],
          },

          {
            name: 'System.Enumerator.Create',
            type: 'sys',
            args: [[new BN(2)]],
          },

          {
            name: 'System.Enumerator.Concat',
            type: 'sys',
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },
        ],
      },
    ],

    result: [new BooleanStackItem(false)],
    gas: FEES[1_000_000],
  },

  {
    name: 'System.Enumerator.Value',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Enumerator.Create',
            type: 'sys',
            args: [[new BN(2)]],
          },

          {
            name: 'System.Enumerator.Create',
            type: 'sys',
            args: [[new BN(1)]],
          },

          {
            name: 'System.Enumerator.Concat',
            type: 'sys',
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },
        ],
      },
    ],

    result: [new IntegerStackItem(new BN(2))],
    gas: FEES[400],
  },

  {
    name: 'System.Enumerator.Value',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Iterator.Create',
            type: 'sys',
            args: [
              {
                type: 'calls',
                calls: [
                  {
                    name: 'NEWMAP',
                    type: 'op',
                  },

                  {
                    name: 'DUP',
                    type: 'op',
                  },

                  {
                    name: 'SETITEM',
                    type: 'op',
                    args: [Buffer.from('value2', 'utf8'), Buffer.from('key2', 'utf8')],
                  },
                ],
              },
            ],
          },

          {
            name: 'System.Iterator.Create',
            type: 'sys',
            args: [
              {
                type: 'calls',
                calls: [
                  {
                    name: 'NEWMAP',
                    type: 'op',
                  },

                  {
                    name: 'DUP',
                    type: 'op',
                  },

                  {
                    name: 'SETITEM',
                    type: 'op',
                    args: [Buffer.from('value1', 'utf8'), Buffer.from('key1', 'utf8')],
                  },
                ],
              },
            ],
          },

          {
            name: 'System.Iterator.Concat',
            type: 'sys',
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },
        ],
      },
    ],

    result: [new BufferStackItem(Buffer.from('value2', 'utf8'))],
    gas: FEES[400],
  },

  {
    name: 'System.Enumerator.Next',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Storage.Find',
            type: 'sys',
            args: [
              {
                type: 'calls',
                calls: [
                  {
                    name: 'System.Storage.GetContext',
                    type: 'sys',
                  },
                ],
              },

              Buffer.alloc(1, 1),
            ],
          },
        ],
      },
    ],

    result: [new BooleanStackItem(true)],
    mockBlockchain: ({ blockchain }) => {
      blockchain.contract.get = jest.fn(async () => Promise.resolve({ manifest: { hasStorage: true } }));

      blockchain.storageItem.getAll$ = jest.fn(() => AsyncIterableX.of(Buffer.alloc(1, 1), Buffer.alloc(1, 2)));
    },
    gas: FEES[1_000_000],
  },

  {
    name: 'System.Enumerator.Next',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Storage.Find',
            type: 'sys',
            args: [
              {
                type: 'calls',
                calls: [
                  {
                    name: 'System.Storage.GetContext',
                    type: 'sys',
                  },
                ],
              },

              Buffer.alloc(1, 1),
            ],
          },
        ],
      },
    ],

    result: [new BooleanStackItem(false)],
    mockBlockchain: ({ blockchain }) => {
      blockchain.contract.get = jest.fn(async () => Promise.resolve({ manifest: { hasStorage: true } }));

      blockchain.storageItem.getAll$ = jest.fn(() => AsyncIterableX.of());
    },
    gas: FEES[1_000_000],
  },

  {
    name: 'System.Enumerator.Value',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'SWAP',
            type: 'op',
            args: [
              {
                type: 'calls',
                calls: [
                  {
                    name: 'System.Enumerator.Next',
                    type: 'sys',
                    args: [
                      {
                        type: 'calls',
                        calls: [
                          {
                            name: 'DUP',
                            type: 'op',
                            args: [
                              {
                                type: 'calls',
                                calls: [
                                  {
                                    name: 'System.Storage.Find',
                                    type: 'sys',
                                    args: [
                                      {
                                        type: 'calls',
                                        calls: [
                                          {
                                            name: 'System.Storage.GetContext',
                                            type: 'sys',
                                          },
                                        ],
                                      },

                                      Buffer.alloc(1, 1),
                                    ],
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      },
    ],

    result: [new BufferStackItem(nextItem.value), new BooleanStackItem(true)],
    mockBlockchain: ({ blockchain }) => {
      blockchain.contract.get = jest.fn(async () => Promise.resolve({ manifest: { hasStorage: true } }));

      blockchain.storageItem.getAll$ = jest.fn(() => AsyncIterableX.of(nextItem));
    },
    gas: FEES[400],
  },

  {
    name: 'System.Enumerator.Value',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Storage.Find',
            type: 'sys',
            args: [
              {
                type: 'calls',
                calls: [
                  {
                    name: 'System.Storage.GetContext',
                    type: 'sys',
                  },
                ],
              },

              Buffer.alloc(1, 1),
            ],
          },

          {
            name: 'System.Iterator.Values',
            type: 'sys',
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },
        ],
      },
    ],

    result: [new BufferStackItem(nextItem.value)],
    mockBlockchain: ({ blockchain }) => {
      blockchain.contract.get = jest.fn(async () => Promise.resolve({ manifest: { hasStorage: true } }));

      blockchain.storageItem.getAll$ = jest.fn(() => AsyncIterableX.of(nextItem));
    },
    gas: FEES[400],
  },

  {
    name: 'System.Enumerator.Value',
    args: [
      {
        type: 'calls',
        calls: [
          {
            name: 'System.Storage.Find',
            type: 'sys',
            args: [
              {
                type: 'calls',
                calls: [
                  {
                    name: 'System.Storage.GetContext',
                    type: 'sys',
                  },
                ],
              },

              Buffer.alloc(1, 1),
            ],
          },

          {
            name: 'System.Iterator.Keys',
            type: 'sys',
          },

          {
            name: 'DUP',
            type: 'op',
          },

          {
            name: 'System.Enumerator.Next',
            type: 'sys',
          },

          {
            name: 'DROP',
            type: 'op',
          },
        ],
      },
    ],

    result: [new BufferStackItem(nextItem.key)],
    mockBlockchain: ({ blockchain }) => {
      blockchain.contract.get = jest.fn(async () => Promise.resolve({ manifest: { hasStorage: true } }));

      blockchain.storageItem.getAll$ = jest.fn(() => AsyncIterableX.of(nextItem));
    },
    gas: FEES[400],
  },
];

describe('SysCalls: System.Iterator & System.Enumerator', () => {
  runSysCalls(SYSCALLS);
});
