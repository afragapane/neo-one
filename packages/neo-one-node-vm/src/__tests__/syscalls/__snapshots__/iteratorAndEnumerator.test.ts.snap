// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 1`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 2`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 3`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 4`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 5`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 6`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 7`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 8`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 9`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 10`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 11`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 12`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 13`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 14`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 15`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 16`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 17`] = `"contract.get"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 18`] = `
Array [
  Array [
    Object {
      "hash": "5bc6631a58b0b0a921bbf909e53e262fe507f144",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 19`] = `"storageItem.getAll$"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 20`] = `
Array [
  Array [
    Object {
      "hash": "5bc6631a58b0b0a921bbf909e53e262fe507f144",
      "prefix": "01",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 21`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 22`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 23`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 24`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 25`] = `"contract.get"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 26`] = `
Array [
  Array [
    Object {
      "hash": "5bc6631a58b0b0a921bbf909e53e262fe507f144",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 27`] = `"storageItem.getAll$"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 28`] = `
Array [
  Array [
    Object {
      "hash": "5bc6631a58b0b0a921bbf909e53e262fe507f144",
      "prefix": "01",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 29`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 30`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 31`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Next 32`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 1`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 2`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 3`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 4`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 5`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 6`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 7`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 8`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 9`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 10`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 11`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 12`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 13`] = `"contract.get"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 14`] = `
Array [
  Array [
    Object {
      "hash": "b2aa50568521792f74796ab14cb1d43f0ac3199a",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 15`] = `"storageItem.getAll$"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 16`] = `
Array [
  Array [
    Object {
      "hash": "b2aa50568521792f74796ab14cb1d43f0ac3199a",
      "prefix": "01",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 17`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 18`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 19`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 20`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 21`] = `"contract.get"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 22`] = `
Array [
  Array [
    Object {
      "hash": "a2a95148e821acda94eeb156b3a0b363e41d1db7",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 23`] = `"storageItem.getAll$"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 24`] = `
Array [
  Array [
    Object {
      "hash": "a2a95148e821acda94eeb156b3a0b363e41d1db7",
      "prefix": "01",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 25`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 26`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 27`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 28`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 29`] = `"contract.get"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 30`] = `
Array [
  Array [
    Object {
      "hash": "fc657bf8b3e2effb062a302002be86f6a47f8e98",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 31`] = `"storageItem.getAll$"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 32`] = `
Array [
  Array [
    Object {
      "hash": "fc657bf8b3e2effb062a302002be86f6a47f8e98",
      "prefix": "01",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 33`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 34`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 35`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Enumerator.Value 36`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Create 1`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Create 2`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Create 3`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Create 4`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 1`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 2`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 3`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 4`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 5`] = `"contract.get"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 6`] = `
Array [
  Array [
    Object {
      "hash": "b2aa50568521792f74796ab14cb1d43f0ac3199a",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 7`] = `"storageItem.getAll$"`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 8`] = `
Array [
  Array [
    Object {
      "hash": "b2aa50568521792f74796ab14cb1d43f0ac3199a",
      "prefix": "01",
    },
  ],
]
`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 9`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 10`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 11`] = `Array []`;

exports[`SysCalls: System.Iterator & System.Enumerator System.Iterator.Key 12`] = `Array []`;
